src/helpers.ts

```diff
@@ -214,10 +214,43 @@ async function getPyrightInfo(): Promise<NpmRegistryResponse> {
     return parseNpmRegistryResponse(JSON.parse(body));
 }
 
-function getPyrightVersion() {
+async function getPyrightVersion() {
     const versionSpec = core.getInput("version");
     if (versionSpec) {
         return new SemVer(versionSpec).format();
     }
+
+    const pylanceVersion = core.getInput("pylance-version");
+    if (pylanceVersion) {
+        if (pylanceVersion !== "latest-release" && pylanceVersion !== "latest-prerelease") {
+            new SemVer(pylanceVersion); // validate version string
+        }
+
+        return await getPylancePyrightVersion(pylanceVersion);
+    }
+
     return "latest";
 }
+
+export interface PylanceBuildMetadata {
+    pylanceVersion: string;
+    pyrightVersion: string;
+}
+
+async function getPylancePyrightVersion(pylanceVersion: string) {
+    const client = new httpClient.HttpClient();
+    const resp = await client.get(
+        `https://raw.githubusercontent.com/microsoft/pylance-release/main/builds/${pylanceVersion}.json`,
+    );
+    const body = await resp.readBody();
+    if (resp.message.statusCode !== httpClient.HttpCodes.OK) {
+        throw new Error(`Failed to download build metadata for Pylance ${pylanceVersion} -- ${body}`);
+    }
+
+    const jsonObject = JSON.parse(body) as PylanceBuildMetadata;
```

Comment:
Can you make this a valita validator rather than casting? See the registry response above for an example.

Reply:
Ah, that's cool. Done.